Add CERN's SWAN to list of deployments

rccern opened the pull request.

minrk committed:
implement null authenticator

rpwagner committed:
Initial SECURITY.md

Proposing a basic security policy, similar to the README or contributors guide, based on the [GitHub documentation](https://docs.github.com/en/code-security/security-advisories/adding-a-security-policy-to-your-repository) and current Project Jupyter recommendations. This may be better as a [default file for the organization](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/creating-a-default-community-health-file).

rickbuilds committed:
adding initial security policy

minrk committed:
make api_page_max_limit configurable

minrk committed:
remove unused Pagination class

used only for 1.4 admin page, not api

minrk committed:
changelog for 2.0

remove suggestions of `admin` permissions, in favor of roles and scopes

Commit:
Add CERN's SWAN to list of deployments

minrk committed:
suggest roles instead of admin_users

and make admin link permission check match admin page

it would be nice if this could be consolidated (maybe an `admin:ui` permission?)

diocas commented on docs/source/gallery-jhub-deployments.md:
```diff
@@ -158,6 +158,14 @@ easy to do with RStudio too.
+  - [Official website](https://cern.ch/swan)
+  - [Spark Monitor](https://github.com/swan-cern/sparkmonitor/) a jupyter extension to monitor Apache Spark applications
+  - [Notebook gallery](https://swan-gallery.web.cern.ch/)
+  - [Sciencebox](https://cern.ch/sciencebox/) a SWAN deployment which includes CERN dependencies
```
```suggestion
  - [Sciencebox](https://cern.ch/sciencebox/) a full analysis setup, with SWAN (based on Zero to JupyterHub), and other CERN components (CERNBox, EOS, CVMFS)
```
@ebocchi what do you suggest? You usually have a nice way of saying things...

minrk committed:
update quickstart requirements

- remove mention of outdated nodejs-legacy
- mention nodesource for more recent node
- mention jupyterlab
- initial localhost request will be on http, not https

minrk committed:
admonition about installing the beta

minrk committed:
expand detail about upgrade revoking tokens

consideRatio committed:
Merge pull request #3602 from minrk/2.0-changelog

2.0 changelog

minrk committed:
2.0.0b1

minrk committed:
publish releases on push to tags

mriedem committed:
Fix 1.3 level

The 1.3 section was a sub-section of 1.4
which makes it harder to find 1.3 release notes
in the changelog from the navigation.

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/psf/black: 21.8b0 → 21.9b0](https://github.com/psf/black/compare/21.8b0...21.9b0)
- [github.com/pre-commit/mirrors-prettier: v2.4.0 → v2.4.1](https://github.com/pre-commit/mirrors-prettier/compare/v2.4.0...v2.4.1)

consideRatio committed:
Merge pull request #3611 from jupyterhub/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

manics committed:
Merge pull request #3610 from mriedem/patch-1

Fix 1.3 level

minrk committed:
add delete scopes for users, groups, servers

e.g. cull-idle services do not need permission to start servers in order to be able to stop them

dependabot[bot] committed:
Bump tmpl from 1.0.4 to 1.0.5 in /jsx

Bumps [tmpl](https://github.com/daaku/nodejs-tmpl) from 1.0.4 to 1.0.5.
- [Release notes](https://github.com/daaku/nodejs-tmpl/releases)
- [Commits](https://github.com/daaku/nodejs-tmpl/commits/v1.0.5)

---
updated-dependencies:
- dependency-name: tmpl
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

manics committed:
Merge pull request #3613 from jupyterhub/dependabot/npm_and_yarn/jsx/tmpl-1.0.5

Bump tmpl from 1.0.4 to 1.0.5 in /jsx

dependabot[bot] committed:
Bump nth-check from 2.0.0 to 2.0.1 in /jsx

Bumps [nth-check](https://github.com/fb55/nth-check) from 2.0.0 to 2.0.1.
- [Release notes](https://github.com/fb55/nth-check/releases)
- [Commits](https://github.com/fb55/nth-check/compare/v2.0.0...v2.0.1)

---
updated-dependencies:
- dependency-name: nth-check
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

minrk committed:
2.0: jupyterlab by default

swaps from default nbclassic and opt-in to lab, to now default to lab and opt-in to nbclassic

defaults to jupyterlab *if* lab 3.1 is available,
so should still work without configuration if lab is unavailable (or too old)

minrk committed:
Merge pull request #3614 from jupyterhub/dependabot/npm_and_yarn/jsx/nth-check-2.0.1

Bump nth-check from 2.0.0 to 2.0.1 in /jsx

manics committed:
Merge pull request #3607 from minrk/recommend-nodesource

update quickstart requirements

manics committed:
nullauthenticator: relative imports, entrypoint, doc

manics committed:
nullauthenticator: missing import

albertmichaelj committed:
Added base_url to path for jupyterhub_session_id cookie

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

minrk committed:
Merge pull request #3625 from albertmichaelj/main

Added base_url to path for jupyterhub-session-id cookie

minrk committed:
Merge pull request #3615 from minrk/lab-by-default

2.0: jupyterlab by default

minrk committed:
server-api example: trim space in token file

avoids invalid newlines in the auth header

consideRatio committed:
Merge pull request #3626 from minrk/token-log

server-api example typo: trim space in token file

consideRatio committed:
Merge pull request #3616 from minrk/delete-scopes

add delete scopes for users, groups, servers

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/asottile/pyupgrade: v2.26.0 → v2.28.0](https://github.com/asottile/pyupgrade/compare/v2.26.0...v2.28.0)

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

manics committed:
Simplify NullAuthenticator, add test

minrk committed:
Merge pull request #3619 from manics/nullauthenticator

Add NullAuthenticator to jupyterhub

minrk committed:
extend deadline for docker build test

it's building 4 images, 10 minutes isn't always enough, bump to 20

minrk committed:
Merge pull request #3627 from jupyterhub/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

minrk committed:
add latest changes to 2.0 changelog

- nullauthenticator
- lab by default

consideRatio committed:
Merge pull request #3628 from minrk/beta-2

add latest changes to 2.0 changelog

minrk committed:
2.0.0b2

minrk committed:
Merge pull request #3445 from rpwagner/patch-1

Initial SECURITY.md

etejedor commented on docs/source/gallery-jhub-deployments.md:
```diff
@@ -158,6 +158,14 @@ easy to do with RStudio too.
 
+### CERN
+
+- SWAN (Service for Web based ANalytics), a platform to perform interactive data analysis in the cloud, integrated with CERN compute, storage and software.
```
Service for Web-based ANalysis
